From 1d3631070612c22f7d8dcb88f9a0bf051c6b1cea Mon Sep 17 00:00:00 2001
From: Marc-Aurel <marc_aurel@me.com>
Date: Mon, 21 Oct 2019 15:51:36 +0200
Subject: [PATCH 8/9] reduced panda safety

---
 panda/board/safety/safety_toyota.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/panda/board/safety/safety_toyota.h b/panda/board/safety/safety_toyota.h
index c1ce996..b60c8ba 100644
--- a/panda/board/safety/safety_toyota.h
+++ b/panda/board/safety/safety_toyota.h
@@ -1,15 +1,15 @@
 // global torque limit
-const int TOYOTA_MAX_TORQUE = 1500;       // max torque cmd allowed ever
+const int TOYOTA_MAX_TORQUE = 3000;       // max torque cmd allowed ever
 
 // rate based torque limit + stay within actually applied
 // packet is sent at 100hz, so this limit is 1000/sec
-const int TOYOTA_MAX_RATE_UP = 10;        // ramp up slow
-const int TOYOTA_MAX_RATE_DOWN = 25;      // ramp down fast
-const int TOYOTA_MAX_TORQUE_ERROR = 350;  // max torque cmd in excess of torque motor
+const int TOYOTA_MAX_RATE_UP = 25;        // ramp up slow
+const int TOYOTA_MAX_RATE_DOWN = 45;      // ramp down fast
+const int TOYOTA_MAX_TORQUE_ERROR = 450;  // max torque cmd in excess of torque motor
 
 // real time torque limit to prevent controls spamming
 // the real time limit is 1500/sec
-const int TOYOTA_MAX_RT_DELTA = 375;      // max delta torque allowed for real time checks
+const int TOYOTA_MAX_RT_DELTA = 1075;      // max delta torque allowed for real time checks
 const uint32_t TOYOTA_RT_INTERVAL = 250000;    // 250ms between real time checks
 
 // longitudinal limits
-- 
2.23.0

